pkg/apis/aws/helper/helper.go

```diff
@@ -81,21 +81,21 @@ func FindSubnetForPurposeAndZone(subnets []api.Subnet, purpose, zone string) (*a
 }
 
 // FindMachineImage takes a list of machine images and tries to find the first entry
-// whose name, version, and zone matches with the given name, version, and region. If no such entry is
+// whose name, version, architecture and zone matches with the given name, version, architecture and region. If no such entry is
 // found then an error will be returned.
-func FindMachineImage(machineImages []api.MachineImage, name, version string) (*api.MachineImage, error) {
+func FindMachineImage(machineImages []api.MachineImage, name, version string, arch *string) (*api.MachineImage, error) {
 	for _, machineImage := range machineImages {
-		if machineImage.Name == name && machineImage.Version == version {
+		if machineImage.Name == name && machineImage.Version == version && *machineImage.Architecture == *arch {
```

Comment:
What if `arch` is `nil`? Some folks might not have updated `gardener/gardener` when updating to this AWS extension version, so defaulting in `CloudProfile` or `Shoot` might not yet be in place.

Reply:
Thanks, I didn't think of this scenario.

Reply:
I don't see this being addressed, or how does it work?

Reply:
Done in this c6870a2f6d57108c294fb0b0aec6a4ff2e075e0e